pepys_import/file/replay_parser.py

```diff
@@ -169,15 +169,14 @@ def process(self, data_store, path, file_contents, data_file_id):
 
                 # and finally store it
                 with data_store.session_scope() as session:
-                    datafile = session.search_datafile_by_id(data_file_id)
-                    platform = session.add_to_platforms_from_rep(
+                    datafile = data_store.search_datafile_by_id(data_file_id)
```

Comment:
thanks for these fixes @BarisSari 

For the earlier one, I hope that Unit Testing will prevent problems like those arising.

But, could you explain what I was doing wrong here?  Was I looking at the wrong object?

The problem wasn't a compiler error, it was a SQLAlchemy error.

Reply:
As I said before, data_store.session_scope() yields objects itself(data_store). So, you needed to call data_store.search_XXX or data_store.add_XXX methods. However, there were other errors causing SQLAlchemy session error. For example, if you look at _add_to_platforms_from_rep_ it returns the cached object, however, it's not in the same session with the current and therefore it gives an error. Therefore, I commented out that lines